_src/lib/node_cache.coffee

```diff
@@ -287,6 +287,26 @@ module.exports = class NodeCache extends EventEmitter
 
 		return delCount
 
+	# ## take
+	#
+	# get the cached value and remove the key from the cache.
+	# Equivalent to calling `get(key)` + `del(key)`.
+	# Useful for implementing `single use` mechanism such as OTP, where once a value is read it will become obsolete.
+	#
+	# **Parameters:**
+	#
+	# * `key` ( String | Number ): cache key
+	#
+	# **Example:**
+	#
+	#	myCache.take "myKey", ( err, val )
+	#
+	take: ( key )=>
+		_ret = @get(key)
+		if (_ret) 
```

Comment:
Wouldn't this "forget" to delete falsy cache values?
I suggest `if (_ret?)`

Reply:
Thanks @erdii  You are right. It does not consider the falsy values. Don't know what I was thinking. The test case was also incomplete - it failed to detect that case.

Added the test case, and fixed the code. Please check now.